src/jit/gentree.cpp

```diff
@@ -11601,13 +11641,14 @@ GenTreePtr          Compiler::gtNewTempAssign(unsigned tmp, GenTreePtr val)
     // struct types. But we don't have a convenient way to do that for all SIMD temps.
 
     CORINFO_CLASS_HANDLE structHnd = gtGetStructHandleIfPresent(val);
-    if (varTypeIsStruct(valTyp) && (structHnd != NO_CLASS_HANDLE))
+    if (varTypeIsStruct(valTyp) && ((structHnd != NO_CLASS_HANDLE) || (varTypeIsSIMD(valTyp))))
```

Comment:
Here too a comment would be useful that we permit simd type nodes with structHdn = null.